### PR TITLE
fix opts maybe nill err

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

File: runc.go [hunk 1]
```diff
@@ -287,7 +287,7 @@ func (r *Runc) Exec(context context.Context, id string, spec specs.Process, opts
 // Run runs the create, start, delete lifecycle of the container
 // and returns its exit status after it has exited
 func (r *Runc) Run(context context.Context, id, bundle string, opts *CreateOpts) (int, error) {
-	if opts.Started != nil {
+	if opts != nil && opts.Started != nil {
```
Comment (AkihiroSuda) [inline runc.go:290]: L309 has the same issue

Comment (AkihiroSuda) [reply runc.go:290]: Probably we should just do `if opts == nil { opts = &CreateOpts} ` and remove all `if opts != nil` checks

Comment (AkihiroSuda) [reply runc.go]: `Exec()` also has the same issue

File: runc.go [hunk 2]
```diff
@@ -287,19 +292,22 @@ func (r *Runc) Exec(context context.Context, id string, spec specs.Process, opts
...
-		if err != nil {
-			return -1, err
-		}
+	oargs, err := opts.args()
+	if err != nil {
+		return -1, err
+	}
+	if len(oargs) > 0 {
 		args = append(args, oargs...)
 	}
```
Comment (austinvazquez) [inline runc.go:308]: non-functional: same here as above. can append without length check.

**Review by austinvazquez — COMMENTED** [review]
Nice work, added some non-functional suggestions for more idiomatic Go / reduces cyclomatic complexity, but overall LGTM.

**Comment by ningmingxiao** [comment]
done  thankyou @austinvazquez

**Review by austinvazquez — APPROVED** [review]
Nice find! LGTM.

**Review by estesp — APPROVED** [review]

**Review by kzys — APPROVED** [review]

**Pull request merged**